### PR TITLE
check the return value of EVP_MD_fetch in ecdh_exch.c & dh_exch.c

### PR DESCRIPTION
**Pull request by x2018**
Check the return value of EVP_MD_fetch in ecdh_exch.c:285 & dh_exch.c:347.

### CODE REVIEW

File: providers/implementations/exchange/dh_exch.c [hunk 1]
```diff
@@ -345,6 +345,8 @@ static int dh_set_ctx_params(void *vpdhctx, const OSSL_PARAM params[])
 
         EVP_MD_free(pdhctx->kdf_md);
         pdhctx->kdf_md = EVP_MD_fetch(pdhctx->libctx, name, mdprops);
+        if (pdhctx->kdf_md == NULL)
+            return 0;
```
Comment (t8m) [inline providers/implementations/exchange/dh_exch.c:349]: ossl_digest_is_allowed handles NULL md gracefully. So this should not be needed as there is another check below. However perhaps somewhat refactoring the code so the duplicate check below is not needed would be even better?

Comment (t8m) [reply providers/implementations/exchange/dh_exch.c:349]: I.E., drop the duplicate check below and add return 0 inside the following `if`.

**Comment by x2018** [comment]
Changed them. However, I still think that although there may be some checks that occur far in the following functions, it is a better habit to check immediately for memory allocation functions as some context relationships are easier omitted during maintenance in the future. By checking in time, we do not need to worry whether the subsequent functions have validity checks.

**Review by paulidale — APPROVED** [review]

**Comment by x2018** [comment]
Sorry, I misunderstood. I should not refactor like the last commit as there are several `if` statements in `ossl_digest_is_allowed()`. If `# if !defined(OPENSSL_NO_FIPS_SECURITYCHECKS)` or `if (ossl_securitycheck_enabled(ctx))` does not satisfy, `pdhctx->kdf_md` would not be checked inside `ossl_digest_is_allowed()`.
And current commit seems to be the version commented by @t8m. I kept the check before the call to `ossl_digest_is_allowed()`, so we can easier know where and why returns 0.

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Squashed and merged to master branch. Thank you for your contribution.

**Pull request closed without merging**